pyrisco/risco.py

```diff
@@ -329,28 +363,50 @@ async def login(self, session=None):
     async def get_state(self):
         """Get partitions and zones."""
         resp = await self._site_post(STATE_URL, {})
+        if resp["state"]["status"]["partitions"] is None:
+            self._partition_panel = False
```

Comment:
Once again, we're repeating the same `if self._partition_panel:` in multiple locations. Let's find an OO way to represent this better.